pkg/util/hostname/os_hostname_linux.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package hostname
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
```

Comment:
This seems to be the crux of things, and my searches for "linux UTS" aren't finding much on /proc/self/uts and various inode numbers.  Could you add a comment here explaining how this works and linking to relevant kernel docs?

Reply:
The explanations are present in https://github.com/DataDog/datadog-agent/blob/91a23189e9c932ad8d3fdc15012743679c86a21f/pkg/util/system/namespace_linux.go